File: src/10844.cpp
//https://www.acmicpc.net/problem/10844
#include <iostream>

using namespace std;

int main() {
	long long answer = 0, n, d[101][10];
	cin >> n;
	d[1][0] = 0;
	for (int i = 1; i < 10; i++)
		d[1][i] = 1;
	
	for (int i = 2; i <= n; i++) {
		for (int j = 0; j < 10; j++) {
			d[i][j] = 0;
			if (j >= 1)		d[i][j] += d[i - 1][j - 1];
			if (j <= 8)		d[i][j] += d[i - 1][j + 1];
			d[i][j] %= 1000000000;
		}
	}
	for (int i = 0; i < 10; i++)
		answer += d[n][i];
	answer %= 1000000000;
	cout << answer << endl;

	return 0;
}

File: src/10844.py
# https://www.acmicpc.net/problem/10844

from sys import stdin


def solution():
    N = int(stdin.readline())
    dp = [[0 for _ in range(10)] for __ in range(N)]
    dp[0] = [0, 1, 1, 1, 1, 1, 1, 1, 1, 1]
    for i in range(1, N):
        for j in range(10):
            if j > 0:
                dp[i][j] += dp[i - 1][j - 1]
            if j < 9:
                dp[i][j] += dp[i - 1][j + 1]
    print(sum(dp[N - 1]) % 1000000000)


solution()
